### PR TITLE
pions

### PR DESCRIPTION
**Pull request by sophiemiddleton**
pion filters for RPC

### CODE REVIEW

**Comment by sophiemiddleton** [comment]
Still need to do some work on the filter to tidy it up

File: EventGenerator/src/RPCGun_module.cc [hunk 1]
```diff
@@ -152,14 +154,12 @@ namespace mu2e {
       const PhysicsParams& gc = *GlobalConstantsHandle<PhysicsParams>();
       double weight = 0.;
       double tau = part->endProperTime() / gc.getParticleLifetime(part->pdgId());
-
       while(part->parent().isNonnull()) { //while particle has a parent which is not null
         part = part->parent();
-        if ( abs(part->pdgId()) == PDGCode::pi_plus ) { //if pion
+        if ( abs(part->pdgId()) == 211 ) { //if pion
```
Comment (kutschke) [inline EventGenerator/src/RPCGun_module.cc:159]: Two things:

1. Always use std::abs, never bare abs.  It leaves you vulnerable to changes in behaviour if transitively included headers change.
2.  Always use symbolic names.  I excpect you can use std::abs(PDGCode::pi_plus).

File: EventGenerator/src/RPCGun_module.cc [hunk 2]
```diff
@@ -169,20 +169,18 @@ namespace mu2e {
     auto output{std::make_unique<StageParticleCollection>()};
     const auto simh = event.getValidHandle<SimParticleCollection>(simsToken_);
     const auto pis = stoppedPiMinusList(simh);
-
+    std::cout<<"event: "<<event.id().event()<<std::endl;
```
Comment (kutschke) [inline EventGenerator/src/RPCGun_module.cc:172]: Please protect this printout with a verbosity flag.

File: EventGenerator/src/PionFilter_module.cc [hunk 3]
```diff
@@ -0,0 +1,99 @@
+// Purpose: Event filter for RPC simulations
```
Comment (kutschke) [inline EventGenerator/src/PionFilter_module.cc:1]: This module should be split up into a filter module and an analyzer that produces the TTree.

**Review by kutschke — CHANGES_REQUESTED** [review]

File: EventGenerator/src/PionFilter_module.cc (hunk 3 above)
Comment (kutschke) [reply EventGenerator/src/PionFilter_module.cc:1]: This means that the isNull parameter will not be needed.

File: EventGenerator/src/RPCGun_module.cc (hunk 2 above)
Comment (sophiemiddleton) [reply EventGenerator/src/RPCGun_module.cc:172]: yes, probably this needs removing.

File: EventGenerator/src/PionFilter_module.cc (hunk 3 above)
Comment (sophiemiddleton) [reply EventGenerator/src/PionFilter_module.cc:1]: I will remove the ntuple, it was for debugging

File: EventGenerator/src/RPCGun_module.cc (hunk 1 above)
Comment (sophiemiddleton) [reply EventGenerator/src/RPCGun_module.cc]: yes, was part of a test due to wierd result, wanted to make sure it was parsing.....fixed now

File: EventGenerator/src/PionFilter_module.cc (hunk 3 above)
Comment (kutschke) [reply EventGenerator/src/PionFilter_module.cc:1]: OK

**Comment by kutschke** [comment]
Please look through the code and fix other instances of abs to std::abs.

**Comment by sophiemiddleton** [comment]
thanks, I found two other instances, now fixed

**Comment by sophiemiddleton** [comment]
@FNALbuild run build test

**Comment by sophiemiddleton** [comment]
Any more comments on this, I would like to begin making pion samples this week

**Review by kutschke — APPROVED** [review]
For a future PR, please split RPCGun_module into a generator, without histograms, and an analyzer that makes histograms.  In the very early days we encouraged a single module that does both but we changed our minds some years ago and we want to move everything in the new direction.  SInce I was not available to pay attention to PRs for so long, I won't insist on it now.

Added issue #1352 to address this.

**Comment by kutschke** [comment]
@FNALbuild run build test

**Comment by kutschke** [comment]
This PR has no coverage in the nightly validation - as best I can tell it affects only JobConfig/beam/POT_infinitepion.fcl .  The oustanding request will be covered in Issue #1352 .  Merging now.

**Pull request merged**